### PR TITLE
Make test-history use batch operations for build and test results

### PR DESCRIPTION
**Pull request by rmmh**
This makes gen_json.py much faster. Now, it spends most of its time
listing directories on GCS and downloading the individual junit xml
files.

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/24221)

### CODE REVIEW

**Comment by spxtr** [comment]
Ooh neat. I'm worried it will run much more slowly on Jenkins, which is what I experienced when I started running it there rather than on my computer.

**Comment by rmmh** [comment]
I'm sure it will run slower there as well, but the difference is hopefully
smaller since it doesn't require thousands of invocations of gsutil.

**Comment by fejta** [comment]
Please rebase and assign back to Joe

**Comment by rmmh** [comment]
Do we want this? The hourly updates rely on being able to talk to Jenkins directly for test results instead of reading from GCS.

File: hack/jenkins/test-history/gen_history [hunk 1]
```diff
@@ -19,10 +19,16 @@
 
 set -o errexit
 set -o nounset
+set -o xtrace
```
Comment (spxtr) [inline hack/jenkins/test-history/gen_history]: Yuck! I think unless we really need it we should avoid xtrace.

Comment (rmmh) [inline hack/jenkins/test-history/gen_history]: Why? It only adds ~10 lines of output to this file, and makes it easier to debug failures.

**Comment by fejta** [comment]
Yes, none of our infrastructure should talk to jenkins direclty.

**Pull request closed without merging**